src/adapters/postgres/cohort-adapter.ts

```diff
@@ -135,12 +138,44 @@ export class PostgresCohortService {
         options: data?.fieldParams?.['options'] || {},
         type: data.type || ''
       };
+      if(data.source_details){
```

Comment:
@Shubham4026 @souravbhowmik1999 This is field specific code, should be in field adapter